src/strategies/math/index.ts

```diff
@@ -12,36 +52,155 @@ export async function strategy(
   options,
   snapshot
 ): Promise<Record<string, number>> {
-  const upstreamResult: Record<string, number> = await strategies[
-    options.strategy.name
-  ].strategy(
-    space,
-    options.strategy.network,
-    getProvider(options.strategy.network),
-    addresses,
-    options.strategy.params,
-    snapshot
+  const strategyOptions: Options = migrateLegacyOptions(options);
+  validateOptions(strategyOptions);
+
+  // Recursively resolve operands
+  const operandPromises: Promise<
+    Record<string, number>
+  >[] = strategyOptions.operands!.map((item) =>
+    resolveOperand(item, addresses, space, snapshot)
+  );
+  const resolvedOperands: Record<string, number>[] = await Promise.all(
+    operandPromises
+  );
+
+  const finalResult: Record<string, number> = resolveOperation(
+    strategyOptions.operation!,
```

Comment:
```
Forbidden non-null assertion
```

Reply:
Code refactored to remove non-null assertion by turning null-able fields into non-null-able ones after validation.